src/main/java/core/basesyntax/model/FruitTransaction.java

```
@@ -0,0 +1,60 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        if (operation == null) {
+            throw new IllegalArgumentException("Operation cannot be null");
+        }
+        if (fruit == null) {
+            throw new IllegalArgumentException("Fruit cannot be null");
+        }
+        if (quantity < 0) {
+            throw new IllegalArgumentException("Quantity cannot be negative");
+        }
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
```

The constructor checks for null `fruit` but does not reject an *empty* or blank fruit name. The task states: "Skip or throw on invalid header or malformed lines (e.g., wrong number of fields, non-integer quantity, empty fruit name, unknown operation code); must not silently ignore without validation." Please validate and reject empty/blank fruit names (for example: `if (fruit.trim().isEmpty()) { throw new RuntimeException("Fruit name cannot be empty"); }`). This prevents transactions with an empty fruit string from being created and later causing unclear behavior.